### PR TITLE
React1 week3

### PR DESCRIPTION
**Pull request by slastushevskyi**

### CODE REVIEW

File: react/react1/week3/vite-project/src/EditComponent.jsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        type="text"
+        value={updatedValue}
+        onChange={(e) => setUpdatedValue(e.target.value)}
+      />
+      <button onClick={(e) => handleUpdate(editedId, e)}>Update</button>
+    </>
+  );
+};
+
+EditComponent.propTypes = {
```
Comment (LucasIversen) [inline react/react1/week3/vite-project/src/EditComponent.jsx:22]: Niiice 👍

File: react/react1/week3/vite-project/src/LifeCycleComponent.jsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import React, { useState, useEffect } from "react";
+
+function countSeconds() {
+  const [counter, setCounter] = useState(0);
+
+  useEffect(() => {
+    const TimerInt =
+      counter >= 0 &&
+      setInterval(() => {
```
Comment (LucasIversen) [inline react/react1/week3/vite-project/src/LifeCycleComponent.jsx:9]: Good

File: react/react1/week3/vite-project/src/TodoItem.jsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+function TodoItem({ item, DeleteTodoItem, CrossOutItem, EditTodoItem }) {
+  return (
+    <li
+      className={item.crossOut ? "notCrossedOut" : "crossedOut"}
```
Comment (LucasIversen) [inline react/react1/week3/vite-project/src/TodoItem.jsx:7]: Nice inline if statement

File: react/react1/week3/vite-project/src/TodoListComponent.jsx [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+function TodoList() {
+  const [todo, setTodo] = useState([]);
+  const [todoId, setTodoId] = useState(4);
+  const [value, setValue] = useState("");
+  const [startDate, setStartDate] = useState(new Date());
+  const formattedDate = moment(startDate).format("YYYY-MM-DD");
+  const [editedId, setEditedId] = useState(-1);
+  const [updatedValue, setUpdatedValue] = useState(" ");
+
+  useEffect(() => {
```
Comment (LucasIversen) [inline react/react1/week3/vite-project/src/TodoListComponent.jsx:21]: Good use of useEffect

File: react/react1/week3/vite-project/src/BorderComponent.jsx [hunk 5]
```diff
@@ -0,0 +1,8 @@
+export default function BorderComponent({ children }) {
+  const BorderComponentStyle = {
+    border: "3px solid black",
+    margin: "10px 0px",
+    padding: "5px",
+  };
+  return <div style={BorderComponentStyle}>{children}</div>;
```
Comment (LucasIversen) [inline react/react1/week3/vite-project/src/BorderComponent.jsx:7]: Perfect use of children!

**Review by LucasIversen — APPROVED** [review]
Hey Nikita. This project is bang on! no changes needed 🥇

File: react/react1/week3/vite-project/src/BorderComponent.jsx (hunk 5 above)
Comment (slastushevskyi) [reply react/react1/week3/vite-project/src/BorderComponent.jsx:7]: Thank you for your review, Lucas. Hope to see you teaching us again ;)

**Pull request merged**